include/alpaka/core/OmpSchedule.hpp

```diff
@@ -0,0 +1,94 @@
+/* Copyright 2020 Sergei Bastrakov
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+
+#ifdef _OPENMP
+#    include <omp.h>
+#endif
+
+#include <cstdint>
+
+
+namespace alpaka
+{
+    namespace omp
+    {
+        //#############################################################################
+        //! Representation of OpenMP schedule information: kind and chunk size.
+        //!
+        //! This class can be used regardless of whether OpenMP is enabled
+        struct Schedule
+        {
+            //#############################################################################
+            //! Mirror of the OpenMP enum omp_sched_t with the corresponding integer values
+            enum Kind
+            {
+                Static = 1,
+                Dynamic = 2,
+                Guided = 3,
+                Auto = 4
+            };
```

Comment:
Why is this not an enum class?

Reply:
So that I can write `Schedule::Static` at construction (in a user code), and not `Schedule::Kind::Static`

Reply:
Why not put the `enum class` outside of the struct? I don't think we can expect many more `kind`s in the `omp` namespace, the specification itself only mentions it with regard to scheduling.

Reply:
Yes, I can put it outside as well. I don't think this makes all that much difference, I just liked most the version of writing (in a user code) smth like `... ompSchedule = alpaka::omp::Schedule{ alpaka::omp::Schedule::Dynamic, 3 };` . We could consider alternatives so that instead of `alpaka::omp::Schedule::Dynamic` there would be something else

Reply:
Okay, I've also realized we can't define our own mirror of `omp_sched_t` enum, since implementations of OpenMP are allowed to extend the standard values with their own. I've now switched to representing as just ints and documented the decision in comments.